app/scripts/services/bundlefactory.js

```
@@ -64,11 +63,13 @@ angular.module('lmisChromeApp')
                 "bundle_lines": getBundleLines(bundleUUID)
               };
               deferred.resolve(bundle);
+            } else {
+              deferred.reject();
```

@jofomah, you can use [$q](http://docs.angularjs.org/api/ng/service/$q)'s `deferred.reject()` when there's no result, rather than returning `null`, then pass an error callback in `.then()` (`then(successCallback, errorCallback, notifyCallback)`) (e.g. [bundleNotFound](https://github.com/eHealthAfrica/LMIS-Chrome/pull/72/files#diff-36a190b2214b4c74c7ddc69df327bbe4R62)).